src/adagrams.js

```diff
@@ -1,8 +1,144 @@
-const Adagrams = {
+class Adagram {
+
   drawLetters() {
-    // Implement this method for wave 1
-  },
-};
+    const letterPoolObject = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4, 
+      E: 12, 
+      F: 2, 
+      G: 3,
+      H: 2, 
+      I: 9,
+      J: 1, 
+      K: 1, 
+      L: 4, 
+      M: 2,
+      N: 6, 
+      O: 8,
+      P: 2, 
+      Q: 1, 
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+    }
+
+    // Add letters to an array
+    let letterPoolArray = [];
+
+    for (let [letter, frequency] of Object.entries(letterPoolObject)) {
+      let i = 0;
+      while (i < frequency) {
+        letterPoolArray.push(letter);
+        i++;
+      }
+    }
+
+    // Select random 10 letters for hand
+    let lettersInHand = [];
+    let j = 0;
+    while (j < 10) {
+      lettersInHand.push(letterPoolArray[Math.floor(Math.random() * letterPoolArray.length)]);
+      j ++;
+    }
+
+    return lettersInHand;
+  }
+
+  usesAvailableLetters(input, lettersInHand) {
+    let allValid = true;
+
+    for (const charIndex in input) {
+      if (lettersInHand.includes(input[charIndex])) {
+        const handIndex = lettersInHand.indexOf(input[charIndex]);
+        lettersInHand.splice(handIndex, 1);
+      } else {
+        allValid = false;
```

Comment:
I believe you could short circuit out of this `for` by `return`ing here, not that it will save you much time in this loop.